### PR TITLE
front: update timestop table

### LINKED ISSUE #9352: front: new output table
### Description and goal

Relates to https://github.com/osrd-project/osrd-confidential/issues/668

The current output table of the scenario page does not match the design system.
We should create a new one (see the mock-up below):

Image

The scope of this ticket is only to create the new output table (and not the buttons below). The table can not be modified (for the moment).

There are still 2 things to clarify with @thibautsailly: what should we display in the column "Reception sur signal fermé" and which title do we want for the new column that @RomainValls is going to add in his PR (https://github.com/OpenRailAssociation/osrd/pull/9275) (will it change the global display ?) ?

### Acceptance criteria

The new output table is in the scenario page

### PR DESCRIPTION
**Pull request by Synar**
Close #9352 

A few points (name of the CI column, width of each column, text boldness and color in the ch column) remain open in the issue, and will be patched in this PR as soon as they are clarified. All other aspects should be implemented.

### CODE REVIEW

**Comment by Synar** [comment]
The last commit hardcode the line breaks and thus split the translations in 2, but I would rather revert this change

**Review by clarani — COMMENTED** [review]
- Can you set the correct background color ? (behind the table)
- the 3 first columns are not in the right order (see the mockup)

image

File: front/src/modules/timesStops/ReadOnlyTime.tsx
Comment (clarani) [inline front/src/modules/timesStops/ReadOnlyTime.tsx]: Can you put all the style in the css ? :) (put `text-right` and `w-full` in `read-only-time`)

Comment (Synar) [reply front/src/modules/timesStops/ReadOnlyTime.tsx]: Sure, done ^^

**Comment by Synar** [comment]
Done ^^

File: front/public/locales/en/timesStops.json
Comment (theocrsb) [inline front/public/locales/en/timesStops.json]: `"realMargin": "real margin (s.)",`

File: front/public/locales/en/timesStops.json
Comment (theocrsb) [inline front/public/locales/en/timesStops.json:18]: `"stopTime": "stopping time (s.)",`

File: front/public/locales/en/timesStops.json
Comment (theocrsb) [inline front/public/locales/en/timesStops.json]: `"theoreticalMarginSeconds": "linear theoretical margin (s.)",`

File: front/public/locales/fr/timesStops.json
Comment (theocrsb) [inline front/public/locales/fr/timesStops.json:1]: Same as for the English file. There are dots after the `(s)`.

File: front/public/locales/fr/timesStops.json
Comment (theocrsb) [inline front/public/locales/fr/timesStops.json]: "theoreticalMarginSeconds": "marge linéaire théorique (s.)",

**Review by theocrsb — CHANGES_REQUESTED** [review]
Thank you for this PR. I've posted a few comments

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (theocrsb) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: The shadow box on the mockup is a little different: 
box-shadow: 0px 1px 0px rgba(255, 255, 255, 1) inset, 0px 4px 7px -3px rgba(255, 171, 88, 0.17), 0px 2px 4px rgba(0, 0, 0, 0.22);

**Comment by theocrsb** [comment]
The name of the output table should be added:
image

File: front/public/locales/en/timesStops.json
Comment (Synar) [reply front/public/locales/en/timesStops.json]: I disagree with this. The SI way to write units like seconds does not take a dot, I don't see why we should differ on this.

File: front/public/locales/fr/timesStops.json
Comment (Synar) [reply front/public/locales/fr/timesStops.json:1]: Same as above.

**Comment by Synar** [comment]
> The name of the output table should be added

I was not sure whether the name was included in the table and thus the desired changes, but I don't mind adding it

**Comment by theocrsb** [comment]
The font size is 14 everywhere except the header, which has a font size of 12px

**Review by theocrsb — CHANGES_REQUESTED** [review]
The font size is 14 everywhere except the header, which has a font size of 12px

File: front/public/locales/en/timesStops.json
Comment (theocrsb) [reply front/public/locales/en/timesStops.json]: in the mockup there is a dot : 
image

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Good catch thanks! How did you get the exact box-shadow css? I couldn't find it on the mockup in sketch.

**Comment by Synar** [comment]
> > The name of the output table should be added
> 
> I was not sure whether the name was included in the table and thus the desired changes, but I don't mind adding it

Thibaut is a bit busy right now so I don't want to bother him for that, but for now I have added a title in line with the information and style already on the page

File: front/public/locales/en/timesStops.json
Comment (Synar) [reply front/public/locales/en/timesStops.json]: This is true, but I believe we should follow the standard unit conventions

**Comment by Synar** [comment]
Good catch on the font size thanks, I did not think to check it

File: front/public/locales/en/timesStops.json
Comment (theocrsb) [reply front/public/locales/en/timesStops.json]: Perhaps we can ask @thibautsailly for confirmation

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (theocrsb) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: right click and `Elevation` :) 
image

File: front/public/locales/fr/simulation.json
Comment (theocrsb) [inline front/public/locales/fr/simulation.json:73]: In the mockup there is:
"timetableOutput": "Table de sortie",

Comment (Synar) [reply front/public/locales/fr/simulation.json:73]: I know as I said above, but there was already a translation key to "table horaire", the title of the corresponding input section is "horaires et arrets", and I don't know if the title in the mockup was set taking the page into account and not just the component, which the title is not really a part of. Same reason why I put the same style as the title already set above.

There is also a button next to it in the mockup, but it's not implemented, and so I don't think it makes sense to try and replicate this part of the mockup now. I think setting properly all the elements and buttons outside the table and uniformizing the sections in the page should probably be another mockup and pass. This is also why I didn't add any title in the first place.

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (theocrsb) [inline front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx:1]: The mockup's widths are not exactly equal to the values used

**Review by theocrsb — CHANGES_REQUESTED** [review]

**Comment by theocrsb** [comment]
I can't see the `CI` anymore when I try to modify a train

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (Synar) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx:1]: The mockup width were supposed to be dynamic. Our columns are fixed width. See the issue for this discussion.

Comment (theocrsb) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx:1]: mb I didn't see

**Comment by Synar** [comment]
> I can't see the `CI` anymore when I try to modify a train

Good catch thanks, fixed, mb

File: front/public/locales/en/timesStops.json
Comment (Synar) [reply front/public/locales/en/timesStops.json]: I did, but in the meantime I think we should defer to the BIPM specifications for units :
> Les symboles d’unités sont des entités mathématiques et non des abréviations. Par conséquent, ils ne doivent pas être suivis d’un point, sauf s’ils se trouvent placés à la fin d’une phrase

File: front/public/locales/fr/timesStops.json
Comment (clarani) [inline front/public/locales/fr/timesStops.json]: We usually order our keys by alphebetical order (maybe we should add an eslint rule for that...)

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Why cell-index for the classname ? Why not .secondary-code-column ?

File: front/src/modules/timesStops/TimesStopsOutput.tsx
Comment (clarani) [inline front/src/modules/timesStops/TimesStopsOutput.tsx]: It's 40px on the mockup
```suggestion
      headerRowHeight={40}
```

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (clarani) [inline front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: Why did you move this function inside the hook ? I think you can leave it outside, this way it won't be re-instantiated at each execution

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (clarani) [inline front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: On the mockup, the margin columns are not aligned on the right but on the left. Same for the arrival column
```suggestion
```

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (clarani) [inline front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]:
```suggestion
              className="ml-1 text-nowrap overflow-hidden"
```

File: front/src/modules/timesStops/TimesStopsOutput.tsx
Comment (Synar) [reply front/src/modules/timesStops/TimesStopsOutput.tsx]: The issue is that in some languages we can get header names that are too long to fit on 2 lines. On the other hand, we could also make the columns wider to ensure they always fit on two lines regardless of the language. The mockup was based on the idea that the column width would adapt to the headers rather than be fixed.

If we do want to instead go for columns dynamically tightly fitted to the content size, I believe we may need to change the library we use for the tables.

File: front/src/modules/timesStops/styles/_readOnlyTime.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_readOnlyTime.scss]: I think that the read-only-time column are align on the left on the mockup.
```suggestion
```

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (Synar) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: I agree, but then on the issue thibaut stated that "All columns but index and CI are right aligned.".

I asked for clarification on this (perhaps he was refering to the headers only? But they also seem to be left aligned on the mockup), but he hasn't had time to answer yet.

Comment (Synar) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: image

File: front/src/modules/timesStops/styles/_readOnlyTime.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_readOnlyTime.scss]: Same as above. While we're waiting for an answer from thibaut I don't mind aligning everything on the left though.

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (Synar) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: I don't remember, so sure.

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (Synar) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: Done, thanks!

I am deeply confused about the difference between text-nowrap and whitespace-nowrap though ^^''

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: I thought of it as the CH code was acting like a secondary index, like the very first index column, hence their similar color and thus the name.

I am fine with changing it to secondary-code-column though ^^

File: front/public/locales/fr/timesStops.json
Comment (Synar) [reply front/public/locales/fr/timesStops.json]: oh right, shortSlipDistance was added in another commit and I put them in the wrong order when resolving the conflict, I'll fix it ^^

File: front/public/locales/en/timesStops.json
Comment (Synar) [reply front/public/locales/en/timesStops.json]: Thibaut confirmed the points were not intentional

**Comment by theocrsb** [comment]
Is the size of the `CI` column as expected? 
image

**Comment by Synar** [comment]
> Is the size of the `CI` column as expected? 
> image
> 

I believe so, see thibaut comment on this in the issue. There should be a title tag to let you read the rest by hovering

**Comment by theocrsb** [comment]
Thibaut said ` CI values have a maximum of 35 characters for now`
I think we don't have 35 characters.
https://github.com/OpenRailAssociation/osrd/issues/9352#issuecomment-2416941708

**Comment by Synar** [comment]
Maximum, not minimum

> • **All columns but CI** have a fixed width based on (...)
> • CI values have a **maximum** of 35 characters for now. (...)
> • When the viewport is too small to display the whole CI value, a **title tag** should carry over. (...)

I have reduced the max width of the CI column btw, though this will be pushed later.
I discussed with thibaut and the title and units are fine as is, but we are going to make the durations right aligned (the times will stay left aligned, both for ease of comparison). 
The headers in English, as well as a couple of French ones, will use abbreviations with title tags with the full values in order to both make column narrowers and reduce the height of the header.
Finally we may change the font of the min/100km unit, or otherwise find a way to make this column narrower.

**Review by theocrsb — COMMENTED** [review]
I don't know if is this PR but we have a weird display in input table : 
image
When we enter 0% we don't see anything displayed, but the array has kept a value.
image

And we see the two points
image

**Comment by Synar** [comment]
The fact that 0% values are hidden was already a hard coded behavior (in 2 different places, so clearly intentional). I don't understand why this is the case. I will discuss this with baptiste as soon as he's back, as well as other questions about margins displaying

Which column is the : with no seconds in?

**Comment by theocrsb** [comment]
> The fact that 0% values are hidden was already a hard coded behavior (in 2 different places, so clearly intentional). I don't understand why this is the case. I will discuss this with baptiste as soon as he's back, as well as other questions about margins displaying
> 
> Which column is the : with no seconds in?

Ok. In `Requested Arrival Time`

**Comment by Synar** [comment]
I will remove the margins changes from this PR, to keep its scope more contained, and add them to another PR later (so until then the 0% will stay hidden, sorry).

I can not reproduce your bug about times being cut. Perhaps the column was simply too narrow? I can change the input table column width to make sure everything fits well.

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (theocrsb) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: This font-size is not applied everywhere
image

**Review by theocrsb — CHANGES_REQUESTED** [review]
I don't think we want to change the style of the input panel. (which would solve the problem of unseen seconds).

File: front/src/modules/timesStops/TimesStopsOutput.tsx
Comment (Synar) [reply front/src/modules/timesStops/TimesStopsOutput.tsx]: We now use abbreviations (with title tags for the full header text) for headers that are too long. The header height was thus adjusted back to 40px.

File: front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx
Comment (Synar) [reply front/src/modules/timesStops/hooks/useTimeStopsColumns.tsx]: After discussion with thibaut, durations should be right-aligned and times should be left-aligned for ease of comparison. Text columns will also be left-aligned by default.

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Fixed. thanks

File: front/src/modules/timesStops/TimesStopsOutput.tsx
Comment (clarani) [reply front/src/modules/timesStops/TimesStopsOutput.tsx]: Ok, thanks for the explanations :pray:

Comment (clarani) [reply front/src/modules/timesStops/TimesStopsOutput.tsx]: Ah hum it's still 50px

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: With `semibold`, I don't see any change :/ I think you can use 600 for now, we will have to add new vars in the css
```suggestion
    font-weight: 600;
```

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Do you think you can encapsulate the css concerning a row in the class dsg-row  ? Like this:
```
.dsg-row {
    &.oddRow {
      --dsg-cell-background-color: var(--ambiantB5);
      --dsg-cell-disabled-background-color: var(--ambiantB5);
    }

    &.activeRow {
      font-weight: 600;
      color: var(--black100);
      .dsg-cell-gutter,
      .secondary-code-column {
        font-weight: normal;
      }
    }
  }
```

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: You can use padding-inline:
```suggestion
    padding-inline: 8px 16px;
```

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Can you use rem instead of px for the font-size ? :) 
```suggestion
    font-size: .875rem;
```

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (clarani) [inline front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Same here
```suggestion
    font-size: .75rem;
```

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: > Ah hum it's still 50px

The last batch of changes isn't pushed yet

> With semibold, I don't see any change

sure

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Sure thanks!

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Sure!

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Done

File: front/src/modules/timesStops/styles/_timesStopsDatasheet.scss
Comment (Synar) [reply front/src/modules/timesStops/styles/_timesStopsDatasheet.scss]: Sure thanks

File: front/public/locales/en/timesStops.json
Comment (clarani) [inline front/public/locales/en/timesStops.json]:
```suggestion
  "departureTimeFull": "requested departure time",
```

**Review by clarani — COMMENTED** [review]
Almost good:
- can you make the column CI take all the remaining space ? So the table fills the whole page ?
image

- can you make the input table fill the whole space too in its tab ?
image

- can you rebase and solve the conflicts ?

**Review by theocrsb — APPROVED** [review]
LGTM and tested! Thank you for this PR

File: front/public/locales/en/timesStops.json
Comment (Synar) [reply front/public/locales/en/timesStops.json]: Oh my bad!

Comment (Synar) [reply front/public/locales/en/timesStops.json]: All done ^^

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Pull request merged**
